### PR TITLE
feat: Add QueryArrowStream to snowflakeConn

### PR DESCRIPTION
**Pull request by zeroshade**
### Description
Added a method to `snowflakeConn` called `QueryArrowStream` which returns a loader object which can be used to retrieve `io.ReadCloser`s which are Arrow IPC streams of record batches of results. This allows a consumer to fetch Arrow data directly without having to rely on using the "exact same" version of the arrow library as this driver does.

This was instrumental in building out the Snowflake ADBC Driver as per https://github.com/apache/arrow-adbc/pull/586

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: chunk_downloader.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ type snowflakeChunkDownloader struct {
 	RowSet             rowSetType
 	FuncDownload       func(context.Context, *snowflakeChunkDownloader, int)
 	FuncDownloadHelper func(context.Context, *snowflakeChunkDownloader, int) error
-	FuncGet            func(context.Context, *snowflakeChunkDownloader, string, map[string]string, time.Duration) (*http.Response, error)
+	FuncGet            func(context.Context, *snowflakeConn, string, map[string]string, time.Duration) (*http.Response, error)
```
Comment (sfc-gh-igarish) [inline chunk_downloader.go:68]: Can we add another method? I believe once interface publish not a good idea to change the method signature.

Comment (zeroshade) [reply chunk_downloader.go:68]: I'm not sure what you mean, this interface isn't a public one, so it shouldn't be an issue to change the signature, right?

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Comment by zeroshade** [comment]
@sfc-gh-igarish would you please re-run the tests, I was unable to replicate the test failure myself but I did see one situation where there's a difference in some logic so I think the last change should fix the issue. If not I'll likely need some help debugging this from you guys since I can't replicate this at all on my end hitting my own snowflake test warehouse.

**Pull request merged**